lasagne/layers/pool.py

```diff
@@ -579,14 +579,18 @@ def __init__(self, incoming, pool_size, stride=None, pad=(0, 0, 0),
     __all__.remove('MaxPool3DLayer')
 
 
-class Upscale1DLayer(Layer):
+class UpscaleLayer(Layer):
     """
-    1D upscaling layer
+    Upscaling layer
 
-    Performs 1D upscaling over the trailing axis of a 3D input tensor.
+    Performs upscaling over the trailing axes of the input tensor.
 
     Parameters
     ----------
+
+    dim: integer
+        Dimension for upscaling. Must be smaller or equal to dimension of input
```

Comment:
The first argument should be `incoming`, to be consistent with all the other layer classes. I'd see `UpscaleLayer` not only as a hidden class to implement `Upscale*DLayer`, but as a layer that can be used by itself.
Furthermore, I'd like to give it an `axes` argument instead of a dimensionality, so you can use it to scale a subset of the dimensions, and even to repeat instances or feature maps.